frontend/src/views/CalculateNutrients/CalculateNutrientsComponents/FertigationModal.tsx

```diff
@@ -404,289 +474,525 @@ export default function FertigationModal({
+                    label="Tank Volume"
+                    value={formData.tankVolume}
+                    onChange={(e) => handleInputChanges({ tankVolume: e })}
                   />
                 </Grid>
-                <Grid size={{ xs: 4 }}>
+                <Grid size={{ xs: 6 }}>
+                  <Select
+                    isRequired
+                    items={TANK_VOLUME_UNITS}
+                    label="Units"
+                    placeholder="Select Units"
+                    selectedKey={formData.tankUnitId}
+                    onSelectionChange={(e) => handleInputChanges({ tankUnitId: e as number })}
+                    noSort
+                    autoselectFirst
+                  />
+                </Grid>
+                <Grid size={{ xs: 6 }}>
                   <NumberField
                     isRequired
-                    label={
-                      <span>
-                        P<sub>2</sub>O<sub>5</sub> (%)
-                      </span>
+                    label="Solubility"
+                    value={formData.solubility}
+                    onChange={(e) => handleInputChanges({ solubility: e })}
+                    minValue={0}
+                  />
+                </Grid>
+                <Grid size={{ xs: 6 }}>
+                  <Select
+                    isRequired
+                    items={SOLUBILITY_RATE_UNITS}
+                    label="Units"
+                    placeholder="Select Units"
+                    selectedKey={formData.solubilityUnitId}
+                    onSelectionChange={(e) => handleInputChanges({ solubilityUnitId: e as number })}
+                    noSort
+                    autoselectFirst
+                  />
+                </Grid>
+                <Grid size={{ xs: 6 }}>
+                  <NumberField
+                    isRequired
+                    label="Amount to Dissolve"
+                    value={formData.amountToDissolve}
+                    onChange={(e) => handleInputChanges({ amountToDissolve: e })}
+                    minValue={0}
+                  />
+                </Grid>
+                <Grid size={{ xs: 6 }}>
+                  <Select
+                    isRequired
+                    items={AMOUNT_TO_DISSOLVE_UNITS}
+                    label="Units"
+                    placeholder="Select Units"
+                    selectedKey={formData.amountToDissolveUnitId}
+                    onSelectionChange={(e) =>
+                      handleInputChanges({ amountToDissolveUnitId: e as number })
                     }
-                    value={formData.customNutrients.P2O5}
-                    onChange={(e) => handleCustomChanges('P2O5', e)}
-                    maxValue={100}
+                    noSort
+                    autoselectFirst
                   />
                 </Grid>
-                <Grid size={{ xs: 4 }}>
+                <Grid size={{ xs: 12 }}>
+                  <div
+                    css={{
```

Comment:
We rly should be putting this css into a styles file instead of sprinkling them into the tsx file.